app/com/yetu/oauth2provider/controllers/OAuth2TrustedServer.scala

```diff
@@ -19,36 +24,38 @@ class OAuth2TrustedServer(scopeService: ScopeService,
    * THIS CODE MUST NOT BE DEPLOYED TO DEMO SERVER !!!!
    * @return
    */
-  def getUserProfile = Action {
+  def getUserProfile = Action.async {
     implicit request =>
-      authorizeTrustedServer {
-        userId =>
-          {
-            val identity = personService.findYetuUser(userId)
-            identity match {
-              case Some(id) => {
-                val outputJson: Option[JsValue] = scopeService.getInfoByScope(id, Config.SCOPE_CONTACT).map(user => Json.toJson(user))
-                outputJson match {
-                  case None           => BadRequest("You have been authorized, but your scope is invalid or not authorized, you cannot access any data for this user.")
-                  case Some(userJson) => Ok(userJson)
-                }
+      authorizeTrustedServer { userId =>
+        {
+
+          personService.findUser(userId).map {
```

Comment:
findUser return a Future[Option[Something]], so a future of option can me used as map { which is the same as map(foo => foo match {})... anyway.. this code will be removed, because this is a security issue currently, someone open sources the project with a trusted server, but the key to the trusted server is public :dancer: